pages/favs.jsx

```diff
@@ -3,18 +3,61 @@ import { getFavs } from '../js/favs-manager';
 import ProjectList from '../components/project-list/project-list.jsx';
 import HintMessage from '../components/hint-message/hint-message.jsx';
 
+import Service from '../js/service.js';
+import config from '../config/config.js';
+import env from "../config/env.generated.json";
+
 export default React.createClass({
   getInitialState() {
     return {
-      favs: null
+      favs: null,
+      loggedInCheckTimestamp: null
     };
   },
+  verifyLoggedInStatus() {
+    Service.nonce()
+      .then((response) => {
+        this.updateLoggedInInfo(true, JSON.parse(response).user);
+      })
+      .catch((reason) => {
+        this.updateLoggedInInfo(false);
+        console.error(reason);
+      });
+  },
+  updateLoggedInInfo(status, username = ``) {
+    config.userLoggedInStatus.set(status);
+    config.username.set(username);
+    this.setState({loggedInCheckTimestamp: new Date()});
+  },
+  logOutUser(event) {
+    event.preventDefault();
+
+    Service.logout()
+      .then(() => {
+        // we have successfully logged user out
+        this.updateLoggedInInfo(false);
+      })
+      .catch((reason) => {
+        console.error(reason);
+      });
+  },
   componentDidMount() {
+    // get IDs of user's favorited entries
     this.setState({favs: getFavs()});
+
+    // let's verify what the logged in status is
+    this.verifyLoggedInStatus();
   },
   render() {
+    let redirectBackUrl = `${env.ORIGIN}${this.props.router.getCurrentLocation().pathname}`;
+    let logInUrl = config.logInUrl.get(redirectBackUrl);
+    let loggedInStatus = config.userLoggedInStatus.get() ?
+                        <p>Hi {config.username.get()}! <a href="" onClick={this.logOutUser}>Log out</a>.</p>
```

Comment:
Don't use an `<a>`  if you're not linking to an actual URL. This is not a hyperlinking navigation element, it's just a button, so let's use the correct markup: `<button tabIndex={0} onClick={..} className={'text-link-button'}>`with the classname imparting the correct styling to make the button look like clickable text in line with the rest of the client.